src/etc/inc/util.inc

```diff
@@ -152,8 +152,10 @@ function gen_subnet($ipaddr, $bits)
 /* return the subnet address given a host address and a subnet bit count */
 function gen_subnetv6($ipaddr, $bits)
 {
-    if (!is_ipaddrv6($ipaddr) || !is_numeric($bits)) {
+    if(!isset($config['system']['dhcpd6track6allowoverride'])) {
```

Comment:
if we need to push such high level override logic into these innocent functions the whole approach is flawed ;)

Reply:
Well if we do it by interface then we'll still have to do this.

Reply:
makes zero sense to me, the caller of gen_subnetv6() should do what is necessary. I'm guessing that's in the radvd or dhcp6 code?

Reply:
The key there is the line at 155, if we don't override it there and let the function continue then we return nothing, and we need a range for dhcp6.conf

Reply:
sure, but that's dhcp's problem, not gen_subnetv6(). this code here merely introduces a multitude of regressions into unrelated code if dhcpd6track6allowoverride were set, like OpenVPN...

Reply:
But it's called by the function that generates dhcpd6.conf, that needs to be updated on a prefix change.

Line 1249 onward in services.inc...

`        if (is_ipaddrv6($ifcfgipv6)) {
            $dhcpdv6conf .= "\nsubnet6 {$subnetv6}/{$ifcfgsnv6}";
        } elseif (!empty($dhcpv6ifconf['range']['from'])) {
            $subnet6 = gen_subnetv6($dhcpv6ifconf['range']['from'], "64");
            $dhcpdv6conf .= "\nsubnet6 {$subnet6}/64";
        }
        $dhcpdv6conf .= " {\n";`

Reply:
I guess I could create a new function specifically for that, then it does not affect anything else.

Reply:
yes and no, we need to address this specifically in "Line 1249 onward in services.inc..."

Reply:
We could use the same lines 161 to 163 in from util.inc - the Net_IPV6:: calls and place them directly into services.inc..

Reply:
sooo you want to bypass the sanity check to generate a subnet anyway? why not

```
function gen_subnetv6($ipaddr, $bits, $validate = true)
```

Reply:
There is no sanity on my desk.. :) Sorry I got distracted setting up my new new Qotom.

OK, so we call the gen_subnetv6() which by default is validate true, but we use the status of the validate flag instead of the config->dhcpd6track6allowoverride. Is that your thought?

Reply:
yep, sounds like a step forward. as far as semantics go "dhcpd6track6allowoverride" doesn't really tell us why we override it so we also need a comment there why we disable validation in this particular case, but first things first...